crates/ledger/src/lib.rs

```diff
@@ -45,16 +43,17 @@ pub mod rewards;
 pub mod state;
 pub mod store;
 
-pub struct Stage<T>
+pub struct Stage<HS, S>
 where
-    T: Store,
+    HS: HistoricalStores,
```

Comment:
Given the prevalence of the word "snapshot" in the codebase and specs for referring to these, should we consider calling it simply `Snapshots`?

Reply:
Sounds good! Do we want to change `ViewStore` to `Snapshot` then?

Reply:
_possibly_; the other option is "ViewStore" / "ReadOnlyStore"; so in one case we name based on _capabilities_ and in the other based on the _responsibilities_. Can't decide which one is better, so whatever you prefer 😅 ...